File: leetcode/0506. Relative Ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        const int n = score.size();
        vector<pair<int, int>> guysToScore;
        for (int i = 0; i < n; ++i) {
            guysToScore.push_back(make_pair(score[i], i));
        }

        sort(guysToScore.rbegin(), guysToScore.rend());

        vector<string> ans(n);
        for(int i = 0; i < n; ++i){
            const int guyInx = guysToScore[i].second;
            switch(i){
                case 0:
                    ans[guyInx] = "Gold Medal";
                    break;
                case 1:
                    ans[guyInx] = "Silver Medal";
                    break;
                case 2:
                    ans[guyInx] = "Bronze Medal";
                    break;
                default:
                    ans[guyInx] = to_string(i + 1);
                    break;
            } 
        }
        return ans;
    }
};

File: leetcode/0506. Relative Ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        ans = [""] * len(score)
        for i, guy_score in enumerate(
            sorted(list(enumerate(score)), key=lambda e: e[1], reverse=True)
        ):
            if i == 0:
                ans[guy_score[0]] = "Gold Medal"
            elif i == 1:
                ans[guy_score[0]] = "Silver Medal"
            elif i == 2:
                ans[guy_score[0]] = "Bronze Medal"
            else:
                ans[guy_score[0]] = str(i + 1)

        return ans